pkg/types/baremetal/validation/platform.go

```diff
@@ -35,14 +39,63 @@ func ValidatePlatform(p *baremetal.Platform, fldPath *field.Path) field.ErrorLis
 
 	if err := validate.IP(p.APIVIP); err != nil {
 		allErrs = append(allErrs, field.Invalid(fldPath.Child("apiVIP"), p.APIVIP, err.Error()))
+	} else if n != nil {
+		if !n.MachineCIDR.Contains(net.ParseIP(p.APIVIP)) {
+			allErrs = append(allErrs, field.Invalid(fldPath.Child("apiVIP"), p.APIVIP, fmt.Errorf("the API VIP is expected to be in %s subnet", n.MachineCIDR.String()).Error()))
+		}
+
+		for _, clusterNetwork := range n.ClusterNetwork {
```

Comment:
machineCIDR is already exclusive from clusterNetwork and serviceNetwork
https://github.com/openshift/installer/blob/5e9cde9c162b313eed46ad964c9211f90e633681/pkg/types/validation/installconfig.go#L148
https://github.com/openshift/installer/blob/5e9cde9c162b313eed46ad964c9211f90e633681/pkg/types/validation/installconfig.go#L117

Reply:
That's helpful. Thanks!